File: Day 1/app/src/main/java/com/akshar/todolist/model/ToDo.java
package com.akshar.todolist.model;

public class ToDo {
    private String title;
    private String description;
    private String date;
    private String time;
    private String id;
    private boolean isCompleted;

    public ToDo() {
    }

    public ToDo(String title, String description, String date, String time, String id, boolean isCompleted) {
        this.title = title;
        this.description = description;
        this.date = date;
        this.time = time;
        this.id = id;
        this.isCompleted = isCompleted;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public boolean isCompleted() {
        return isCompleted;
    }

    public void setCompleted(boolean completed) {
        isCompleted = completed;
    }
}

File: Day 3/lib/model/todo.dart
class ToDo {
  int id;
  String title;
  String description;
  bool completed;

  ToDo({
    required this.id,
    required this.title,
    required this.description,
    this.completed = false,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'completed': completed ? 1 : 0,
    };
  }

  static ToDo fromMap(Map<String, dynamic> map) {
    return ToDo(
      id: map['id'],
      title: map['title'],
      description: map['description'],
      completed: map['completed'] == 1 ? true : false,
    );
  }

  ToDo copyWith({
    int? id,
    String? title,
    String? description,
    bool? completed,
  }) {
    return ToDo(
      id: id ?? this.id,
      title: title ?? this.title,
      description: description ?? this.description,
      completed: completed ?? this.completed,
    );
  }
}
